### PR TITLE
Tup#229: refactor product creation flow

### PR DESCRIPTION
**Pull request by craxrev**

### CODE REVIEW

**Review by gdalyy — CHANGES_REQUESTED** [review]

File: controllers/MarketplaceOverviewController.php [hunk 1]
```diff
@@ -161,16 +163,24 @@ public function actionNew()
                 throw new HttpException(401);
             }
 
+            if ($model->account == '0') {
```
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/MarketplaceOverviewController.php:166]: In this step (2nd), the user selects which account he wants to use (account select field), the first index "0" is filled manually in L#141, the rest is filled L#143 through the foreach loop.. When saving, we check which account is selected and the logic continues below

File: controllers/ProductController.php [hunk 2]
```diff
@@ -66,34 +62,32 @@ public function actionNew($marketplaceId)
...
-                    $spacesList[$space->id] = SpaceImage::widget(['space' => $space, 'width' => 16, 'showTooltip' => true, 'link' => true]) . ' ' . $space->name;
+                    $accountsList[$space->id] = SpaceImage::widget(['space' => $space, 'width' => 16, 'showTooltip' => true, 'link' => true]) . ' ' . $space->name;
             }
 
-            return $this->renderAjax('../product/spaces-list', [
-                'product' => $model,
-                'spacesList' => $spacesList,
-            ]);
-        }
+            $model->account = 0;
```
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/ProductController.php:81]: this make the first option selected in the account field

File: controllers/ProductController.php [hunk 3]
```diff
@@ -102,22 +96,29 @@ public function actionNew($marketplaceId)
                 throw new HttpException(401);
             }
 
+            if ($model->account == '0') {
```
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/ProductController.php:99]: same logic

File: controllers/ProductController.php [hunk 4]
```diff
@@ -130,21 +131,6 @@ public function actionNew($marketplaceId)
 
             return $this->redirect($url);
         }
-
-        // Check validation
```
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/ProductController.php:134]: there were no validations in this controller, I think validations are handled inside the model, also I thought it is redundant piece of code with no utility so it was deleted/refactored

File: controllers/MarketplaceOverviewController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (gdalyy) [reply controllers/MarketplaceOverviewController.php:166]: Can you use some const for this ?

File: controllers/ProductController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (gdalyy) [reply controllers/ProductController.php:81]: Use const please

File: controllers/ProductController.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (gdalyy) [reply controllers/ProductController.php:134]: Did you try to remove html validation an check your your controller without this deleted part ?

**Review by gdalyy — CHANGES_REQUESTED** [review]

File: controllers/MarketplaceOverviewController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/MarketplaceOverviewController.php:166]: yes

File: controllers/ProductController.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (craxrev) [reply controllers/ProductController.php:134]: I'll check this tomorrow

**Review by gdalyy — APPROVED** [review]

**Pull request merged**